### PR TITLE
5991 Enable lazy resampling for SpatialResample

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
A part of #5991 .

### Description

This PR adds the changes of `SpatialResample` from #5860 .

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/spatial/array.py [hunk 1]
```diff
@@ -237,86 +200,12 @@ def __call__(
         Set `dst_affine` and `spatial_size` to `None` to turn off the resampling step.
         """
         # get dtype as torch (e.g., torch.float64)
-        _dtype = get_equivalent_dtype(dtype or self.dtype or img.dtype, torch.Tensor)
-        align_corners = self.align_corners if align_corners is None else align_corners
+        dtype_pt = get_equivalent_dtype(dtype or self.dtype or img.dtype, torch.Tensor)
+        align_corners = align_corners if align_corners is not None else self.align_corners
```
Comment (yiheng-wang-nv) [inline monai/transforms/spatial/array.py:204]: This change is just to align with the following two lines.

**Review by wyli — APPROVED** [review]

File: monai/transforms/inverse.py [hunk 2]
```diff
@@ -122,7 +122,8 @@ def push_transform(self, data, *args, **kwargs):
             if do_transform:
                 xform = data.pending_operations.pop()
                 xform.update(transform_info)
-                meta_obj = self.push_transform(data, transform_info=xform, lazy_evaluation=lazy_eval)
+                extra = xform.copy()
+                meta_obj = self.push_transform(data, transform_info=xform, lazy_evaluation=lazy_eval, extra_info=extra)
```
Comment (wyli) [inline monai/transforms/inverse.py:126]: Hi @yiheng-wang-nv,
this branch uses:
```py
extra = xform.copy()
xform.update(transform_info)
```
but the prototype branch uses:
```py
xform.update(transform_info)
extra = xform.copy()
```
they are different, is this a typo or a bug fix?

Comment (yiheng-wang-nv) [reply monai/transforms/inverse.py:126]: Thanks @wyli for correction. It was a typo

**Pull request merged**